crates/model/src/order.rs

```diff
@@ -84,6 +85,73 @@ impl Order {
         token_list.contains(&self.creation.buy_token)
             || token_list.contains(&self.creation.sell_token)
     }
+
+    /// Returns the remaining amounts for the order.
+    ///
+    /// For fill-or-kill orders, this trivially returns full buy, sell and fee
+    /// amounts for the order (since the order can only either be not executed
+    /// at all or fully executed).
+    ///
+    /// For partially fillable orders, it considers the currently executed
+    /// amount and computes what is left in the order.
+    ///
+    /// Returns `None` on overflows.
```

Comment:
Hmm... good point.

Originally, I wanted to distinguish between "bad overflows" (that happen if the order we get doesn't make sense, like if the executed amount is higher than the full amount of the order) and "other overflow".

That being said, I think you are right that we can just use errors here since the API shouldn't ever accept orders that can't be traded (i.e where the fill scaling computation in the SC would overflow - making it effectively untradable). And using `Result` instead of `Result<Option>` makes the code more succinct.

Will change.

Reply:
Implemented to return errors on overflow (since they are "bad" cases anyway).